examples/jsm/nodes/utils/RotateNode.js

```diff
@@ -12,23 +13,43 @@ class RotateNode extends TempNode {
 
 	constructor( positionNode, rotationNode ) {
 
-		super( 'vec3' );
+		super();
 
 		this.positionNode = positionNode;
 		this.rotationNode = rotationNode;
 
 	}
 
-	setup() {
+	setup( builder ) {
 
 		const { rotationNode, positionNode } = this;
 
-		const rotation = rotationNode;
-		const rotationXMatrix = mat4( vec4( 1.0, 0.0, 0.0, 0.0 ), vec4( 0.0, cos( rotation.x ), sin( rotation.x ).negate(), 0.0 ), vec4( 0.0, sin( rotation.x ), cos( rotation.x ), 0.0 ), vec4( 0.0, 0.0, 0.0, 1.0 ) );
-		const rotationYMatrix = mat4( vec4( cos( rotation.y ), 0.0, sin( rotation.y ), 0.0 ), vec4( 0.0, 1.0, 0.0, 0.0 ), vec4( sin( rotation.y ).negate(), 0.0, cos( rotation.y ), 0.0 ), vec4( 0.0, 0.0, 0.0, 1.0 ) );
-		const rotationZMatrix = mat4( vec4( cos( rotation.z ), sin( rotation.z ).negate(), 0.0, 0.0 ), vec4( sin( rotation.z ), cos( rotation.z ), 0.0, 0.0 ), vec4( 0.0, 0.0, 1.0, 0.0 ), vec4( 0.0, 0.0, 0.0, 1.0 ) );
+		const nodeType = positionNode.getNodeType( builder );
 
-		return rotationXMatrix.mul( rotationYMatrix ).mul( rotationZMatrix ).mul( vec4( positionNode, 1.0 ) ).xyz;
+		this.nodeType = nodeType;
```

Comment:
I think it would be better to use this pattern to dynamically change types:

https://github.com/mrdoob/three.js/blob/e8884ade569071ccd5bac30dc820306b3ba39e69/examples/jsm/nodes/core/VaryingNode.js#L30-L36